packages/fxa-content-server/app/tests/spec/views/sign_in_password.js

```diff
@@ -236,6 +237,66 @@ describe('views/sign_in_password', () => {
     });
   });
 
+  describe('onSignInError', () => {
+    let loginSubmitErrorStub;
+
+    beforeEach(() => {
+      loginSubmitErrorStub = sinon.stub(GleanMetrics.login, 'error');
+    });
+
+    afterEach(() => {
+      loginSubmitErrorStub.restore();
+    });
+
+    describe('metrics', () => {
+      it('submits an "account locked" login_submit_frontend_error Glean ping', () => {
+        sinon
+          .stub(view, 'signIn')
+          .callsFake(() => Promise.reject(AuthErrors.ERRORS.ACCOUNT_RESET));
+        view.$('#password').val('password');
+
+        return Promise.resolve(view.validateAndSubmit()).then(() => {
```

Comment:
nit: To avoid promise chains can you make the test async and do `await Promise.resolve()...`?